packages/table/src/table.tsx

```diff
@@ -338,6 +336,50 @@ export class Table extends AbstractComponent<ITableProps, ITableState> {
         selectionModes: SelectionModes.ALL,
     };
 
+    private static SHALLOWLY_COMPARABLE_PROP_KEYS = [
```

Comment:
Is there a better way to do this than this? Are there going to be issues in the future as we add more props and they're not in this list? I know there's no way to pull the list of items out of the type because the type isn't real... but this seems like a not-so-great solution. Wouldn't it be better to specify the ones we _don't_ want to do this with?